syntax = "proto3";

package auth;

option go_package = "./auth";

service AuthService {
  rpc DeliverTokenByRPC(DeliverTokenReq) returns (DeliveryResp) {}
  rpc VerifyTokenByRPC(VerifyTokenReq) returns (VerifyResp) {}
  rpc RenewTokenByRPC(RenewTokenReq) returns (RenewTokenResp) {}
}

message DeliverTokenReq {
  int32 user_id = 1;
}

message VerifyTokenReq {
  string token = 1;
}

message RenewTokenReq {
  string token = 1;
}

message DeliveryResp {
  string token = 1;
}

message VerifyResp {
  bool res = 1;
}

message RenewTokenResp {
  string new_token = 1;
}
